lib/index.js

```diff
@@ -41,11 +41,22 @@ var connect = (function connectWrapper() {
     function checkConnected() {
         if (!isConnected) {
             connectionError(new Error("Connection closed prematurely, please check your credentials"));
+        } else if (connectOpts.reconnect){
+            // monkey patch b/c node-amqp seems to be not maintained anymore based
+            // upon number of open issues, comments within issues stating it is no
+            // longer maintained and the age of open pull requests. node-amqp only
+            // publishes an error on the "end" event if the "ready" event has not
+            // yet been fired which means you probably have an issue authing with
+            // Rabbit. They should also have an else block and look to see if the
+            // socket is destroyed and if so, emit an error to which node-amqp
+            // will attempt to reconnect using the specified strategy.
+            connection.emit("error", new Error("Connection closed, attempting to reconnect"));
         }
     }
 
     return function connect(url, opts) {
         if (!connectionPromise) {
+            connectOpts = opts;
```

Comment:
the error handler should be `connection.on("close", checkConnected)` to handle multiple disconnects from Rabbit, otherwise after your first reconnect, it will not attempt to reconnect.